### PR TITLE
refactor(pages): solve compile warnings and remove commented out code

### PR DESCRIPTION
**Pull request by jsvgoncalves**
Submitted-by: @Chreggii

### CODE REVIEW

File: frontend/src/pages/arrange_components/arrange_components.js [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
-require('dotenv').config();
+import '../../../node_modules/react-resizable/css/styles.css';
+import '../../pages.css';
+import 'react-grid-layout/css/styles.css';
+
+import { faQuestion } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import axios from 'axios';
+import React from 'react';
+import { confirmAlert } from 'react-confirm-alert';
```
Comment (jsvgoncalves) [inline frontend/src/pages/arrange_components/arrange_components.js:9]: Is there a reason not to follow this import order?

1. stdlib imports;
2. 3rd party imports;
3.  local/library specific imports

File: frontend/src/pages/arrange_components/arrange_components.js [hunk 2]
```diff
@@ -1,27 +1,16 @@
...
 
-const ResponsiveGridLayout = WidthProvider(Responsive);
+import VisualComponentsLayout from './visual_components_layout';
 
-const Main = styled.main`
```
Comment (jsvgoncalves) [inline frontend/src/pages/arrange_components/arrange_components.js:17]: Is this totally unused across the project?

File: frontend/src/pages/arrange_components/visual_components_layout.js [hunk 3]
```diff
@@ -1,41 +1,30 @@
...
+import Grid from '@material-ui/core/Grid';
+import axios from 'axios';
+import _ from 'lodash';
+import React, { Suspense } from 'react';
+import { confirmAlert } from 'react-confirm-alert';
+import RGL, { WidthProvider } from 'react-grid-layout';
+import Loadable from 'react-loadable';
+import { withRouter } from 'react-router-dom';
+
+import { ToolBox } from './toolbox';
```
Comment (jsvgoncalves) [inline frontend/src/pages/arrange_components/visual_components_layout.js:16]: Same about import order.

File: frontend/src/pages/arrange_components/visual_components_layout.js [hunk 4]
```diff
@@ -122,7 +109,7 @@ class VisualComponentsLayout extends React.PureComponent {
                 const visCompName = JSON.parse(localStorage.getItem("apiResponse")).componentsParameters[compIndex].name;
                 let finalOutput = JSON.parse(localStorage.getItem("fullComponentsInfo"));
                 const finalOutputComps = finalOutput.configuration['1'].components;
-                finalOutputComps.map(v => {
+                finalOutputComps.forEach(v => {
```
Comment (jsvgoncalves) [inline frontend/src/pages/arrange_components/visual_components_layout.js:112]: Why?

File: frontend/src/pages/set_components/checkbox_list.js [hunk 5]
```diff
@@ -8,31 +8,34 @@ export default class CheckboxList extends React.Component {
...
                 let json = JSON.parse(localStorage.getItem("visualComponents"));
-                Object.keys(json).map((e, i) => {
+                Object.keys(json).forEach((e, i) => {
                     this.state[e] = json[e];
                 });
             }
             else {
-                props.values.map((v, i) => {
+                props.values.forEach((v, i) => {
+                    // eslint-disable-next-line
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/checkbox_list.js:17]: Why?

File: frontend/src/pages/set_components/set_components.js [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
-import {confirmAlert} from "react-confirm-alert";
-import listReactFiles from 'list-react-files'
-import "../../pages.css"
-import Grid from "@material-ui/core/Grid";
+import React from 'react';
+import { confirmAlert } from 'react-confirm-alert';
+import { Editors } from 'react-data-grid-addons';
+
+import Settings from './settings';
+
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/set_components.js:12]: Same about import order.

File: frontend/src/pages/set_components/settings.js [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
-import SettingsComponents from "./settings_components";
-import axios from "axios";
-import "../../pages.css"
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withRouter } from 'react-router';
 
-require('dotenv').config();
+import SettingsComponents from './settings_components';
+import SettingsDecisionCards from './settings_decision_cards';
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/settings.js:8]: Same about import order

File: frontend/src/pages/set_components/settings_decision_cards.js [hunk 8]
```diff
@@ -1,10 +1,12 @@
...
-import "../../pages.css"
+import '../../pages.css';
+import './settings_components.css';
+
+import Grid from '@material-ui/core/Grid';
+import React from 'react';
+import ReactDataGrid from 'react-data-grid';
+import Select from 'react-select';
+
+import Checkbox from './checkbox';
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/settings_decision_cards.js:9]: same about import order

File: frontend/src/pages/settings/general_settings.js [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
-import "./general_settings.css";
-import "../../pages.css"
+
+import { faQuestion } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Grid from '@material-ui/core/Grid';
+import axios from 'axios';
+import React from 'react';
+import { confirmAlert } from 'react-confirm-alert';
+import { withRouter } from 'react-router';
```
Comment (jsvgoncalves) [inline frontend/src/pages/settings/general_settings.js:11]: same about import order

File: frontend/src/webpage.js [hunk 10]
```diff
@@ -1,28 +1,28 @@
...
+    updateGenericTimeseries,
+    updateGenericValue,
+    updateRollspecificGoals,
+} from './actions';
+import Logo from './images/logo_nfp73_en.png';
+import ArrangeComponents from './pages/arrange_components/arrange_components';
+import CreateDecisionCard from './pages/create_decision_cards/create_decision_card';
+import Home from './pages/home/home';
+import SetComponents from './pages/set_components/set_components';
+import GeneralSettings from './pages/settings/general_settings';
```
Comment (jsvgoncalves) [inline frontend/src/webpage.js:25]: same about import order

File: frontend/src/pages/set_components/settings_components.js [hunk 11]
```diff
@@ -1,27 +1,21 @@
...
+import '../../pages.css';
+import './settings_components.css';
 
+import Grid from '@material-ui/core/Grid';
+import axios from 'axios';
+import React from 'react';
+import ReactDataGrid from 'react-data-grid';
+import Select from 'react-select';
+
+import Checkbox from './checkbox';
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/settings_components.js:10]: import order

File: frontend/src/pages/set_components/settings_components.js [hunk 12]
```diff
@@ -221,47 +207,22 @@ class SettingsComponents extends React.Component {
         const currCompName = currState.currComponentName;
         const finalOutput = JSON.parse(localStorage.getItem("fullComponentsInfo"));
         const finalOutputComps = finalOutput.configuration['1'].components;
-        let comp_index = 0;
-        finalOutputComps.map(v => {
+        finalOutputComps.forEach(v => {
             if (v.name === currCompName) {
                 v.parameter = gridRows;
                 //this.setState({"currentParameters": v.parameter});
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/settings_components.js:213]: Can remove this as well?
```suggestion

```

File: frontend/src/pages/set_components/settings_components.js [hunk 13]
```diff
@@ -221,47 +207,22 @@ class SettingsComponents extends React.Component {
...
-                                    const response = this.getValueFromSource(apiRequest, parameters, param_index, v.name,
-                                        finalOutputComps, finalOutput, gridRows, comp_index);
-                                }
-                            })
-                        }
-                        catch (e) {}
-                    }
-                    param_index++;
+
+                v.parameter.forEach(() => {
```
Comment (jsvgoncalves) [inline frontend/src/pages/set_components/settings_components.js:216]: Why the change? Is it equivalent?

Comment (jsvgoncalves) [reply frontend/src/pages/set_components/settings_components.js:216]: (It's hard to have an overview, so it probably makes sense what you did, I'm just curious)

File: frontend/src/pages/arrange_components/arrange_components.js (hunk 1 above)
Comment (Chreggii) [reply frontend/src/pages/arrange_components/arrange_components.js:9]: I didn't do it on purpose. I use an extension that sorts that automatically.
1. css files
2. everything which is located in node_modules folder
3. local specific imports

If you like, I can change it :)

File: frontend/src/pages/arrange_components/arrange_components.js (hunk 2 above)
Comment (Chreggii) [reply frontend/src/pages/arrange_components/arrange_components.js:17]: Yes, I think so. 

There is just one "Main" component used and this one is definied in the component itself.
image

File: backend/configuration/configuration_api.py [hunk 14]
```diff
@@ -57,13 +58,12 @@ def clone_git_repo_and_store_path_in_database(git_repo_address, database, local_
     :return:                            {Dictionary}    {'success':True} if everything worked, {'success':False} otherwise
     """
     try:
-        controller = Controller(git_repo_address,
-                                os.path.dirname(os.path.abspath(__file__)) + os.getenv(local_repo_path_env_string))
+        controller = Controller(git_repo_address, Path(os.path.dirname(os.path.abspath(__file__)) + os.getenv(local_repo_path_env_string)))
```
Comment (jsvgoncalves) [inline backend/configuration/configuration_api.py:61]: In general, I'd prefer shorter lines. ~79
That's more or less standard, see e.g. in Python [PEP8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length)

File: backend/configuration/configuration_model.py [hunk 15]
```diff
@@ -82,6 +85,11 @@ def __init__(self, local_repo_path, clone_url):
                 # if git repo in local repo path is not the same repo as given in the clone url
                 if not current_clone_url == clone_url:
                     # remove all files form folder and clone new git repo from given clone url
+                    for root, dirs, files in os.walk(self.local_repo_path):
```
Comment (jsvgoncalves) [inline backend/configuration/configuration_model.py:88]: What was the issue here?

File: frontend/src/pages/set_components/settings_components.js (hunk 13 above)
Comment (Chreggii) [reply frontend/src/pages/set_components/settings_components.js:215]: Yes, I know. The variables are defined, but nowhere used. Therefore, it's possible to remove all those code. Because of the reason, that the "response" varibale is nowhere used, all these staff can be removed.

image

File: backend/configuration/configuration_api.py (hunk 14 above)
Comment (Chreggii) [reply backend/configuration/configuration_api.py:61]: I will change this :)

File: backend/configuration/configuration_model.py (hunk 15 above)
Comment (Chreggii) [reply backend/configuration/configuration_model.py:88]: This code es needed to remove the ".git" folder in windows. `shutil.rmtree` is not enough to remove read-only files

**Comment by jsvgoncalves** [comment]
@Chreggii LGTM, let me know when to merge. (We don't need to immediately address everything like line length etc)

**Comment by Chreggii** [comment]
All comments are resolved :)
